src/Helpers/Pipeline.php

```diff
@@ -77,10 +72,21 @@ public function setPipes(array $pipes): static
     /**
      * Get all the pipes in the pipeline
      *
-     * @return array
+     * @return array<\Saloon\Data\Pipe>
      */
     public function getPipes(): array
     {
         return $this->pipes;
     }
+
+    /**
+     * Check if a given pipe exists for a name
+     *
+     * @param string $name
+     * @return bool
+     */
+    protected function pipeExists(string $name): bool
+    {
+        return ! empty(array_filter($this->pipes, static fn (Pipe $pipe) => $pipe->name === $name));
```

Comment:
I'd change to a foreach here, so you can bail immediately.
If someone has many pipes in the pipeline, it can drastically improve performance.
We really only care if it has one, so there's no need to continue iterating.

```suggestion
        foreach ($this->pipes as $pipe) {
            if ($pipe->name === $name) {
                return true;
            }
        }
        
        return false;
```

Reply:
Awesome, thanks @juse-less !